### ISSUE TITLE
New version of AHF changed the log file name

### ISSUE THREAD

**Issue by rennehan** [yt-project/yt#2227]
### Bug report

**Bug summary**

The newest version of the AHF halo finder (v1.0-100) has a different log file name than what is implemented in the yt code. I am not sure which version of AHF changed this, but the latest has the change.

**Code for reproduction**

Line 83 of yt/frontends/ahf/data_structures.py currently says:

```python
self.log_filename = root + '.log'
```

but it now should be:

```python
self.log_filename = root + '.00.log'
```

Obviously, it would be simple to change the name of the log file, but to be consistent with AHF it may be better for yt to change or at least try to determine if this file might exist (maybe try either?).

**Version Information**

  * Python Version: 3.6.8
  * yt version: 3.5.1

I installed via conda.

**Comment by ngoldbaum** [comment]
I think this probably just needs a try/except block somewhere in the AHF frontend. Pull requests are very welcome :)

If you can share a new AHF output one of us could also try to implement it.

**Comment by rennehan** [comment]
Indeed, I could probably implement that and share a pull request. 

The format of the files is still the same, just the name has changed. So, for example, I have a snapshot_189.hdf5 file that I ran AHF on with the file prefix "snapshot_189.hdf5". This produces a file:

snapshot_189.hdf5.00.log

Whereas before, as I assume from the code in data_structures.py, it would have output:

snapshot_189.hdf5.log

As I said above, the format of the log file is exactly the same, only the name has changed.

**Comment by ngoldbaum** [comment]
Sure, I just suggested to share an output because I don’t like making changes to yt without some “ground-truth” data to test with.

**Comment by rennehan** [comment]
For sure, the two example datasets will be fine:

http://yt-project.org/data/gizmo_64.tar.gz
http://yt-project.org/data/ahf_halos.tar.gz

I am also now thinking that this may occur because I compiled AHF with MPI, but only used 1 computing core. I think it may add the "00" if MPI is compiled in, regardless of how many cores are used. Usually each task gets its own log file however. I'm not sure if yt allows for multiple logs/files in combination with AHF however.

**Comment by ngoldbaum** [comment]
I’m confused, do the two example datasets have this issue? When I say “ground-truth” I mean a real dataset, not one I’ve hacked to emulate a real
one.

Not sure about the MPI issue. Maybe the author of the frontend @qobilidop can share more?

**Comment by rennehan** [comment]
I see, I figured since it was changing a file name it wouldn't be an issue. I can provide the snapshot data, but it is large (~6GB). I don't have any smaller tests at the moment.

**Comment by qobilidop** [comment]
@rennehan I have seen the same thing when AHF is compiled with MPI. Unfortunately, this is not supported in AHF yet.

One workaround is to rename the AHF output. In case of multiple files, they could be concatenated to be a single valid file, because as I observed only the `.00` file contains a header.

**Comment by rennehan** [comment]
Indeed, I simply edited the line in yt to read my proper file. I suppose this is a feature request then, since yt works fine with the serial AHF version but not the MPI version, without modification.

If I end up writing something for this, I will be sure to send a pull request.

**Comment by qobilidop** [comment]
That would be awesome!

**Comment by brittonsmith** [comment]
@rennehan, I'm also happy to help with this if you're able to supply a halo catalog with the `.00` naming convention that currently isn't working.

**Comment by rennehan** [comment]
@brittonsmith @ngoldbaum thanks, I have some data for testing here:

https://drive.google.com/open?id=1zT5FccNR3EVk7UHzlMcbayEyW5Ds1aFA

@qobilidop did you implement reading multiple halo files generated by AHF-MPI?

**Comment by qobilidop** [comment]
@rennehan No, I didn't.
